jbpm-services/jbpm-services-ejb/jbpm-services-ejb-timer/src/main/java/org/jbpm/services/ejb/timer/EJBTimerScheduler.java

```diff
@@ -124,24 +132,32 @@ public boolean removeJob(JobHandle jobHandle) {
 	}
 	
 	public TimerJobInstance getTimerByName(String jobName) {
+	    
+	    if (localCache.containsKey(jobName)) {
+	        logger.debug("Found job {} in cache returning", jobName);
+	        return localCache.get(jobName);
+	    }
+	    TimerJobInstance found = null;
+	    
 		for (Timer timer : timerService.getTimers()) {
 		    try {
     			Serializable info = timer.getInfo();
     			if (info instanceof EjbTimerJob) {
     				EjbTimerJob job = (EjbTimerJob) info;
     				
     				EjbGlobalJobHandle handle = (EjbGlobalJobHandle) job.getTimerJobInstance().getJobHandle();
+    				localCache.putIfAbsent(jobName, handle.getTimerJobInstance());
     				if (handle.getUuid().equals(jobName)) {
     					logger.debug("Job  {} does match timer and is going to be returned", jobName);
-    					return handle.getTimerJobInstance();
+    					found = handle.getTimerJobInstance();
```

Comment:
Nope, the point here is to go over all items only once and that's why it does not break directly